sandy/fy.py

```diff
@@ -422,23 +422,32 @@ def apply_qmatrix(self, zam, energy, decay_data):
         >>> decay_minimal = sandy.get_endf6_file("jeff_33", 'decay', zam)
         >>> decay_fytest = sandy.DecayData.from_endf6(decay_minimal)
         >>> npfy = Fy(minimal_fytest_2)
-        >>> npfy_pert = npfy.apply_bmatrix(942390, 5.00000e+05, decay_fytest)
-        >>> diff = npfy_pert.data[npfy_pert.data.FY != npfy.data.FY].FY
-        >>> comp = npfy_pert.data.query('ZAM==942390 & MT==459 & E==500e3').squeeze().FY
-        >>> assert comp.values.all() == diff.values.all()
-
-        """
-        new_data = self.data.copy()
+        >>> npfy_pert = npfy.apply_qmatrix(942390, 5.00000e+05, decay_fytest)
+        >>> npfy_pert.data[npfy_pert.data.MT == 459]
+             MAT	 MT	   ZAM	   ZAP	          E	         FY	        DFY
+        3	9437	459	942390	591480	5.00000e+05	1.00000e-01	0.00000e+00
+        4	9437	459	942390	591481	5.00000e+05	2.00000e-01	0.00000e+00
+        5	9437	459	942390	601480	5.00000e+05	6.00000e-01	0.00000e+00
+        6	9437	459	942390	621480	5.00000e+05	6.00000e-01	0.00000e+00
+        """
+        # Obtain the data:
+        data = self.data.copy()
         conditions = {'ZAM': zam, 'MT': 454, "E": energy}
-        fy_data = self._filters(conditions).data.set_index('ZAP')['FY']
-        index = fy_data.index
+        fy_data = self._filters(conditions).data
+        mat = fy_data.iloc[0, 0]
```

Comment:
it is better `mat = fy_data.MAT.squeeze()`

Reply:
Dont work, i thought is because the have different shapes

Reply:
well, at least use `fy_data.MAT.iloc[0]`